driver-mapping/src/main/java/com/datastax/driver/mapping/MethodMapper.java

```diff
@@ -92,6 +88,30 @@ public void prepare(MappingManager manager, PreparedStatement ps) {
         }
     }
 
+    // Checks the method parameters against the query's bind variables
+    private void validateParameters() {
+        if (method.isVarArgs())
+            throw new IllegalArgumentException(String.format("Invalid varargs method %s in @Accessor interface", method.getName()));
+
+        ColumnDefinitions variables = statement.getVariables();
+        Set<String> names = Sets.newHashSet();
+        for (ColumnDefinitions.Definition variable : variables) {
+            names.add(variable.getName());
+        }
+
+        if (method.getParameterTypes().length < names.size())
```

Comment:
:+1: Thanks for the explanation. (although I feel that the error messages below might be difficult to interpret, it would be more helpful if we could be more specific than just saying "at least" or "at most" - but I can live with that.)